move-vm-backend/src/abi.rs

```diff
@@ -0,0 +1,244 @@
+use alloc::borrow::ToOwned;
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use move_binary_format::access::ModuleAccess;
+use move_binary_format::file_format::{
+    Ability, AbilitySet, SignatureToken, StructFieldInformation, StructHandleIndex, Visibility,
+};
+use move_binary_format::CompiledModule;
+use move_core_types::account_address::AccountAddress;
+use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::ModuleId;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize, Ord, PartialOrd, Eq, PartialEq)]
+pub struct ModuleAbi {
```

Comment:
Done

Reply:
> Struct and StructDef structures in ABI.

I left a TODO comment for this case since I'm not sure if we can merge those yet. Let's come back to this question once we start handling the RPC request in the upper layers to see how the ABI output looks like